app/components/cases/cases-header.hbs

```diff
@@ -18,6 +18,16 @@
             >
               {{t "add-case"}}
             </AuButton>
+            {{#if this.enableNewCaseCreation}}
```

Comment:
We might as well put this behind a permission instead of putting this button inside the previous permission check. It makes it easier to remove the old one later.
It also prevents accidentally releasing this without permissions.
`(user-may "manage-cases")`  is fine for now.